core/src/test/scala/unit/kafka/server/ListOffsetsRequestTest.scala

```diff
@@ -202,7 +210,17 @@ class ListOffsetsRequestTest extends BaseRequestTest {
 
   @ParameterizedTest
   @ValueSource(strings = Array("zk", "kraft"))
-  def testResponseIncludesLeaderEpoch(quorum: String): Unit = {
+  def testResponseIncludesLeaderEpochWithUnstableAPIs(quorum: String): Unit = {
+    testResponseIncludesLeaderEpoch()
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResponseIncludesLeaderEpochWithStableAPIs(quorum: String): Unit = {
+    testResponseIncludesLeaderEpoch()
+  }
+  
+  def testResponseIncludesLeaderEpoch(): Unit = {
```

Comment:
I am not sure if this truly captures the problem. The test seems to verify that the broker can accept the latest version of ListOffset when enableUnstableVersions is set to false. This means that we could never introduce an unstable version of ListOffset in the future. When introducing a new version of ListOffset, it's ok to make it unstable, as long as the stable MV doesn't use it for inter broker communication.

Reply:
Maybe we can have a IT used to make sure MV does not return "unstable" version for inter broker communication. for example:
```java
    @ClusterTests({
        @ClusterTest(serverProperties = {@ClusterConfigProperty(key = ServerConfigs.UNSTABLE_API_VERSIONS_ENABLE_CONFIG, value = "false")}),
        @ClusterTest(serverProperties = {@ClusterConfigProperty(key = ServerConfigs.UNSTABLE_API_VERSIONS_ENABLE_CONFIG, value = "true")})
    })
    public void test(ClusterInstance clusterInstance) {
        clusterInstance.brokers().values().forEach(b -> {
            MetadataVersion mv = b.metadataCache().metadataVersion();
            Assertions.assertTrue(mv.listOffsetRequestVersion() <= ApiKeys.LIST_OFFSETS.latestVersion(false));
            Assertions.assertTrue(mv.fetchRequestVersion() <= ApiKeys.FETCH.latestVersion(false));
            Assertions.assertTrue(mv.offsetForLeaderEpochRequestVersion() <= ApiKeys.OFFSET_FOR_LEADER_EPOCH.latestVersion(false));
        });
    }
```

Reply:
@chia7712 : Good idea. We probably don't need an IT. We could just use the latest production MV like the following.

```
    public void test() {
        MetadataVersion mv = MetadataVersion.latestProduction()
        Assertions.assertTrue(mv.listOffsetRequestVersion() <= ApiKeys.LIST_OFFSETS.latestVersion(false));
        Assertions.assertTrue(mv.fetchRequestVersion() <= ApiKeys.FETCH.latestVersion(false));
        Assertions.assertTrue(mv.offsetForLeaderEpochRequestVersion() <= ApiKeys.OFFSET_FOR_LEADER_EPOCH.latestVersion(false));
        });
    }
```

Reply:
@junrao you always have better solution :)

Reply:
`TransactionMarkerChannelManager` sets the version for `WriteTxnMarkersRequest` based on MV in the following code in `TransactionMarkerChannelManager`. It would be useful to move that logic to MetadataVersion and add a similar test on unstable version. This could be done in a separate PR if it requires more work.

```
  private val writeTxnMarkersRequestVersion: Short =
    if (config.interBrokerProtocolVersion.isAtLeast(IBP_2_8_IV0)) 1
    else 0
```

Reply:
>  This could be done in a separate PR if it requires more work.

I feel following small changes can address it, so this PR should be able to include it. @m1a2st WDYT?

```java
    public short writeTxnMarkersRequestVersion() {
        if (isAtLeast(IBP_2_8_IV0)) {
            return 1;
        } else {
            return 0;
        }
    }
```

Reply:
Yes, I will address these comments in this PR.